ci-operator/step-registry/openshift/microshift/infra/iso-boot/openshift-microshift-infra-iso-boot-commands.sh

```diff
@@ -98,15 +98,36 @@ trap 'finalize' EXIT
 # by the time we try to kill them. There is only 1 child now, but this is generic enough to allow N.
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} || true; wait; fi' TERM
 
-SCENARIO_SOURCES="/home/${HOST_USER}/microshift/test/scenarios"
+# Determine the tests to run depending on the job name and type.
+# Exclude long-running tests from presubmit jobs.
 EXCLUDE_CNCF_CONFORMANCE=false
+if [ "${JOB_TYPE}" == "presubmit" ]; then
+  EXCLUDE_CNCF_CONFORMANCE=true
+fi
+
+# Implement scenario directory check with fallbacks. Simplify or remove the
+# function when the structure is homogenised in all the active releases.
+function get_source_dir() {
+  local -r base="/home/${HOST_USER}/microshift/test"
+  local -r ndir="$1"
+  local -r fdir="$2"
+
+  if [ -e "${base}/${ndir}" ] && [ -d "${base}/${ndir}" ] ; then
+    echo "${base}/${ndir}"
```

Comment:
I'm not sure if this would be successful after merging the other change.
We're checking the paths that are present on the metal host, but this code is running inside the CI container.

Reply:
Should I use SSH to test for directory existence?

Reply:
In iso-build we package the sources:
```
tar czf /tmp/microshift.tgz /go/src/github.com/openshift/microshift
```

So maybe instead of ~/microshift use the other path?

Reply:
What about the current remote check to keep the script self-contained? It executes the same ssh command below to run the procedure.